src/main/java/at/ac/tuwien/damap/conversion/AbstractTemplateExportScienceEuropeComponents.java

```diff
@@ -813,6 +818,7 @@ public void composeTableNewDatasets(XWPFTable xwpfTable){
         }
         //end of dynamic table rows code
         xwpfTable.removeRow(1);
+        replaceTableVariables(xwpfTable, replacements);
```

Comment:
Tables are composed and filled in a general function. So we should either call `replaceTableVariables` in each specific compose method (as is done here) or just after calling the compose method (as is done here https://github.com/tuwien-csd/damap-backend/blob/ecead07507490d78a3c50e72030c1085c4784ce1/src/main/java/at/ac/tuwien/damap/conversion/AbstractTemplateExportScienceEuropeComponents.java#L749)

Reply:
<img width="713" alt="image" src="https://github.com/tuwien-csd/damap-backend/assets/80054001/bbdafde2-670e-4292-8524-b1aba32192d4">

Unfortunately, this instance of `replaceTableVariables` cannot be called inside the specific compose method, since it is meant to fill tables that do not have a specific compose method. 

An alternative would be to have the compose methods return a boolean, to indicate if the table was deleted or not. Then we can call `replaceTableVariables` after the switch statement.

Reply:
Ah right, we remove tables now. Will it throw, if we remove a table and then call `replaceTableVariables` on it?

Reply:
Yes, the problem here is, that removing a table from a document removes the underlying xml, but the java obejct stays. replaceTableVariables tries to access the text (xml), which causes an exception. I have now changed it so that replaceTableVariables is called after all tables are composed and only for tables that still exist.